stl/inc/yvals.h

```diff
@@ -178,6 +178,14 @@ _STL_DISABLE_CLANG_WARNINGS
 #error _ITERATOR_DEBUG_LEVEL != 0 must imply _CONTAINER_DEBUG_LEVEL == 1.
 #endif // _ITERATOR_DEBUG_LEVEL != 0 && _CONTAINER_DEBUG_LEVEL == 0
 
+#if _CONTAINER_DEBUG_LEVEL > 0
+#define _TEMPLATE_CDL  template <int _Cdl = _CONTAINER_DEBUG_LEVEL>
+#define _COMMA_INT_CDL , int _Cdl = _CONTAINER_DEBUG_LEVEL
+#else // _CONTAINER_DEBUG_LEVEL > 0
+#define _TEMPLATE_CDL

```

Comment:
I really don't understand why we are only making these templates sometimes. They are always templated entities, they can never be address taken, etc. whether they are member templates or not. Why not just make them unconditionally templates if we're going to do this?

Reply:
I'm worried about disrupting code that does strange things like export classes derived from Standard classes.

Reply:
Given our current explicit lack of support for doing that, I don't think we should be inconsistent about attempting to support it here. (And if anything, templatizing all these hides them from dllimport/dllexport which is usually what the customer wants in such broken cases anyway)